<a href="https://colab.research.google.com/github/Sebastian4524/Datavisualization/blob/main/pandas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **1. Library**

In [1]:
import numpy as np
import pandas as pd

# **2. Import a dataset**

In [2]:
s = "/content/datosp.xlsx"
d = pd.read_excel(s)

# **3. Exploration**

In [3]:
d.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 759 entries, 0 to 758
Data columns (total 16 columns):
 #   Column                                        Non-Null Count  Dtype  
---  ------                                        --------------  -----  
 0   Region                                        759 non-null    object 
 1   Country code                                  759 non-null    object 
 2   Economy                                       759 non-null    object 
 3   Reporting year                                759 non-null    int64  
 4   Survey name                                   759 non-null    object 
 5   Survey year                                   759 non-null    int64  
 6   Survey coverage                               755 non-null    object 
 7   Welfare type                                  755 non-null    object 
 8   Survey comparability                          755 non-null    float64
 9   Monetary (%)                                  759 non-null    flo

In [4]:
d.head()

Region Country code  Economy  Reporting year Survey name  Survey year  \
0    SSA          AGO   Angola            2018       IDREA         2018   
1    ECA          ALB  Albania            2012        LSMS         2012   
2    ECA          ALB  Albania            2016         HBS         2016   
3    ECA          ALB  Albania            2017         HBS         2017   
4    ECA          ALB  Albania            2018         HBS         2018   

  Survey coverage Welfare type  Survey comparability  Monetary (%)  \
0               N            c                   2.0     31.122005   
1               N            c                   1.0      0.620691   
2               N            c                   2.0      0.139303   
3               N            c                   2.0      0.392570   
4               N            c                   3.0      0.048107   

   Educational attainment (%) Educational enrollment (%) Electricity (%)  \
0                   29.753423                   27.44306       52.639532   
1                    2.271381                   6.731328        0.496704   
2                    0.622464                          -        0.431298   
3                    0.331231                          -        0.131322   
4                    0.192380                          -         0.06025   

  Sanitation (%) Drinking water (%)  \
0      53.637516          32.106507   
1       3.040233           0.342548   
2       7.603243           8.402681   
3       6.976078            9.27045   
4       6.579772           9.594966   

   Multidimensional poverty headcount ratio (%)  
0                                     47.203606  
1                                      0.792693  
2                                      0.781341  
3                                      0.711346  
4                                      0.293161

In [5]:
d.tail()

Region Country code       Economy  Reporting year Survey name  \
754    MNA          YEM   Yemen, Rep.            2014         HBS   
755    SSA          ZAF  South Africa            2014         LCS   
756    SSA          ZMB        Zambia            2015    LCMS-VII   
757    SSA          ZWE      Zimbabwe            2017       PICES   
758    SSA          ZWE      Zimbabwe            2019       PICES   

     Survey year Survey coverage Welfare type  Survey comparability  \
754         2014               N            c                   1.0   
755         2014               N            c                   3.0   
756         2015               N            c                   5.0   
757         2017               N            c                   0.0   
758         2019               N            c                   0.0   

     Monetary (%)  Educational attainment (%) Educational enrollment (%)  \
754     19.802757                   15.950564                  44.500461   
755     20.501289                    2.263442                   2.338686   
756     60.793096                   24.374674                    30.3689   
757     34.409153                    6.893227                   3.100654   
758     39.754530                    0.927010                    5.98423   

    Electricity (%) Sanitation (%) Drinking water (%)  \
754       33.891296      41.205913           14.01668   
755        4.093847      35.171476          10.361161   
756       69.213849       59.97659          34.368798   
757       37.062222      67.637062          22.405361   
758        37.99479       38.28523          19.260145   

     Multidimensional poverty headcount ratio (%)  
754                                     35.411829  
755                                     21.710479  
756                                     66.403395  
757                                     39.897895  
758                                     42.397930

# **4. Cleaning**

## 4.1. Column names

In [6]:
d.rename(columns = {"Reporting year":"year"}, inplace = True)
d.rename(columns = {"Region":"region"}, inplace = True)
d.rename(columns = {"Survey comparability":"survey comparability"}, inplace = True)
d.rename(columns = {"Monetary (%)":"monetary"}, inplace = True)
d.rename(columns = {"Educational attainment (%)":"ea"}, inplace = True)
d.rename(columns = {"Educational enrollment (%)":"ee"}, inplace = True)
d.rename(columns = {"Electricity (%)":"electricity"}, inplace = True)
d.rename(columns = {"Sanitation (%)":"sanitation"}, inplace = True)
d.rename(columns = {"Drinking water (%)":"dwater"}, inplace = True)
d.rename(columns = {"Multidimensional poverty headcount ratio (%)":"multidimensional ratio"}, inplace = True)

In [7]:
d2=d

## 4.2. Delete columns

In [8]:
d2 = d2.drop(columns = ["Survey coverage","Welfare type","Survey name","survey comparability"])

In [9]:
d2.head()

region Country code  Economy  year  Survey year   monetary         ea  \
0    SSA          AGO   Angola  2018         2018  31.122005  29.753423   
1    ECA          ALB  Albania  2012         2012   0.620691   2.271381   
2    ECA          ALB  Albania  2016         2016   0.139303   0.622464   
3    ECA          ALB  Albania  2017         2017   0.392570   0.331231   
4    ECA          ALB  Albania  2018         2018   0.048107   0.192380   

         ee electricity sanitation     dwater  multidimensional ratio  
0  27.44306   52.639532  53.637516  32.106507               47.203606  
1  6.731328    0.496704   3.040233   0.342548                0.792693  
2         -    0.431298   7.603243   8.402681                0.781341  
3         -    0.131322   6.976078    9.27045                0.711346  
4         -     0.06025   6.579772   9.594966                0.293161

## 4.3. Drop NA

In [10]:
d2 = d2.replace("-",np.NaN)
d2.head()

region Country code  Economy  year  Survey year   monetary         ea  \
0    SSA          AGO   Angola  2018         2018  31.122005  29.753423   
1    ECA          ALB  Albania  2012         2012   0.620691   2.271381   
2    ECA          ALB  Albania  2016         2016   0.139303   0.622464   
3    ECA          ALB  Albania  2017         2017   0.392570   0.331231   
4    ECA          ALB  Albania  2018         2018   0.048107   0.192380   

          ee  electricity  sanitation     dwater  multidimensional ratio  
0  27.443060    52.639532   53.637516  32.106507               47.203606  
1   6.731328     0.496704    3.040233   0.342548                0.792693  
2        NaN     0.431298    7.603243   8.402681                0.781341  
3        NaN     0.131322    6.976078   9.270450                0.711346  
4        NaN     0.060250    6.579772   9.594966                0.293161

In [11]:
d2 =d2.dropna()

In [12]:
d2.head()

region Country code    Economy  year  Survey year   monetary         ea  \
0    SSA          AGO     Angola  2018         2018  31.122005  29.753423   
1    ECA          ALB    Albania  2012         2012   0.620691   2.271381   
5    LAC          ARG  Argentina  2010         2010   0.749329   1.904169   
6    LAC          ARG  Argentina  2011         2011   0.707419   1.807899   
7    LAC          ARG  Argentina  2012         2012   0.640789   1.650053   

          ee  electricity  sanitation     dwater  multidimensional ratio  
0  27.443060    52.639532   53.637516  32.106507               47.203606  
1   6.731328     0.496704    3.040233   0.342548                0.792693  
5   0.963598     0.000000    0.683219   0.325656                0.772191  
6   1.051406     0.000000    0.716557   0.308643                0.743913  
7   0.833707     0.000000    0.523555   0.324075                0.675336

In [13]:
d2["year_str"] = d2 ["year"].astype(str)

<ipython-input-13-f047fb21a0aa>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  d2["year_str"] = d2 ["year"].astype(str)


# **5. Indexing**

In [14]:
d2 = d2.set_index(["Economy","year"])
d2.head()

region Country code  Survey year   monetary         ea  \
Economy   year                                                          
Angola    2018    SSA          AGO         2018  31.122005  29.753423   
Albania   2012    ECA          ALB         2012   0.620691   2.271381   
Argentina 2010    LAC          ARG         2010   0.749329   1.904169   
          2011    LAC          ARG         2011   0.707419   1.807899   
          2012    LAC          ARG         2012   0.640789   1.650053   

                       ee  electricity  sanitation     dwater  \
Economy   year                                                  
Angola    2018  27.443060    52.639532   53.637516  32.106507   
Albania   2012   6.731328     0.496704    3.040233   0.342548   
Argentina 2010   0.963598     0.000000    0.683219   0.325656   
          2011   1.051406     0.000000    0.716557   0.308643   
          2012   0.833707     0.000000    0.523555   0.324075   

                multidimensional ratio year_str  
Economy   year                                   
Angola    2018               47.203606     2018  
Albania   2012                0.792693     2012  
Argentina 2010                0.772191     2010  
          2011                0.743913     2011  
          2012                0.675336     2012

# **6. Descriptive Statistics**

In [15]:
# Basic
d2[["ea","ee","electricity","sanitation","dwater","multidimensional ratio"]].describe()

ea          ee  electricity  sanitation      dwater  \
count  336.000000  336.000000   336.000000  336.000000  336.000000   
mean    13.554299    6.946490    12.455075   20.977328    9.780775   
std     15.986560   10.372906    22.527316   24.115456   11.667525   
min      0.000000    0.055241     0.000000    0.000000    0.000000   
25%      3.224699    1.297253     0.231926    3.212466    1.562115   
50%      6.302084    2.846354     1.599842   11.082474    5.690938   
75%     17.102751    6.311659     9.913953   29.749310   12.326958   
max     75.149274   58.042753    97.250843   95.897704   69.206844   

       multidimensional ratio  
count              336.000000  
mean                14.225384  
std                 20.937838  
min                  0.000000  
25%                  1.327811  
50%                  5.093471  
75%                 16.907301  
max                 85.678405

In [24]:
import pandas as pd

# Define una función personalizada para el formato
def custom_format(x):
    if isinstance(x, float):
        return '{:.3f}'.format(x)
    return str(x)

# Aplica la función de formato personalizada a todos los datos del DataFrame
pd.options.display.float_format = custom_format

# Ejecuta tu código
d2[["ea", "ee", "electricity", "sanitation", "dwater", "multidimensional ratio"]].describe()

ea      ee  electricity  sanitation  dwater  multidimensional ratio
count 336.000 336.000      336.000     336.000 336.000                 336.000
mean   13.554   6.946       12.455      20.977   9.781                  14.225
std    15.987  10.373       22.527      24.115  11.668                  20.938
min     0.000   0.055        0.000       0.000   0.000                   0.000
25%     3.225   1.297        0.232       3.212   1.562                   1.328
50%     6.302   2.846        1.600      11.082   5.691                   5.093
75%    17.103   6.312        9.914      29.749  12.327                  16.907
max    75.149  58.043       97.251      95.898  69.207                  85.678

In [17]:
d2[["ea", "ee" , "sanitation", "dwater", "multidimensional ratio"]].describe().applymap(lambda x:"{:,.1f}".format(x))

ea     ee sanitation dwater multidimensional ratio
count  336.0  336.0      336.0  336.0                  336.0
mean    13.6    6.9       21.0    9.8                   14.2
std     16.0   10.4       24.1   11.7                   20.9
min      0.0    0.1        0.0    0.0                    0.0
25%      3.2    1.3        3.2    1.6                    1.3
50%      6.3    2.8       11.1    5.7                    5.1
75%     17.1    6.3       29.7   12.3                   16.9
max     75.1   58.0       95.9   69.2                   85.7

# **7. Conclusions**

<ul>
  <li>The country that has the highest educational attainment is 75.1 % meaning that just 24.9% of their population does not have a degree</li>
  <li>the ceros on the descriptive data might be that the country could be so poor that they can not measure their sanitation or drinking water </li>
  <li>We can see that the highest multidimentional ratio is a combination of the other high variables </li>
</ul>


In [25]:
d2[["dwater"]].groupby("year").describe().applymap(lambda x:"{:,.1f}".format(x))

dwater                                          
      count  mean   std   min   25%   50%   75%   max
year                                                 
2009    1.0  69.2   nan  69.2  69.2  69.2  69.2  69.2
2010   25.0  11.9  13.1   0.2   2.4   7.8  17.3  50.6
2011   21.0  12.8  14.2   0.1   2.0   7.7  18.2  56.0
2012   31.0  14.6  15.8   0.3   2.6   8.7  22.6  59.9
2013   23.0   7.0   7.4   0.3   1.2   5.2   8.7  25.6
2014   26.0  13.1  12.6   0.4   4.5   9.2  20.0  44.9
2015   34.0  11.2  12.3   0.0   2.1   8.0  11.0  42.7
2016   33.0   9.6  10.9   0.0   1.6   4.7  12.8  40.8
2017   29.0   5.9   6.4   0.1   0.7   3.9   8.7  22.4
2018   42.0  11.9  11.6   0.1   2.0   7.3  20.9  37.5
2019   31.0   5.9   6.5   0.0   0.7   3.1   9.0  23.7
2020   16.0   3.9   2.7   0.1   1.6   4.3   5.7   8.6
2021   17.0   3.9   3.7   0.3   0.7   3.7   5.3  13.6
2022    7.0   2.6   2.3   0.0   0.8   2.6   4.0   6.0

# **8. Conclusions**

<ul>
  <li>with the pass of de tiempo we can see tht in the world the drinking water in decreasing for everyone</li>
  <li>the year in which we were able to see the highest metric for the cuantity of people that have access to drinking wated was in 2009 </li>
  <li>we can see that the worst year for the drinking water metric was in 2022  </li>
</ul>

In [53]:
# by country
Eco = d2.filter(like = "Argentina", axis = 0)
Argentina.head()

region Country code  Survey year  monetary    ea    ee  \
Economy   year                                                          
Argentina 2010    LAC          ARG         2010     0.749 1.904 0.964   
          2011    LAC          ARG         2011     0.707 1.808 1.051   
          2012    LAC          ARG         2012     0.641 1.650 0.834   
          2013    LAC          ARG         2013     0.623 1.619 0.896   
          2014    LAC          ARG         2014     0.612 1.418 0.949   

                electricity  sanitation  dwater  multidimensional ratio  \
Economy   year                                                            
Argentina 2010        0.000       0.683   0.326                   0.772   
          2011        0.000       0.717   0.309                   0.744   
          2012        0.000       0.524   0.324                   0.675   
          2013        0.000       0.406   0.321                   0.657   
          2014        0.000       0.399   0.406                   0.635   

               year_str  
Economy   year           
Argentina 2010     2010  
          2011     2011  
          2012     2012  
          2013     2013  
          2014     2014

In [51]:
# by year
y2009 =d2.query("year == 2009")
y2009.head()

region Country code  Survey year  monetary     ea    ee  \
Economy          year                                                           
Papua New Guinea 2009    EAP          PNG         2009    39.683 22.216 9.004   

                       electricity  sanitation  dwater  \
Economy          year                                    
Papua New Guinea 2009       82.574      79.779  69.207   

                       multidimensional ratio year_str  
Economy          year                                   
Papua New Guinea 2009                  74.658     2009

# **9. Pivot**


In [52]:
## Be careful you must use the un indexed dataframe

## exaple
## rows = year
## columns = country code
## values = pop_g

pivot1 = d.pivot(index = "year",
                 columns = "Economy",
                 values = "ea")
pivot1.head()

Economy  Albania  Angola  Argentina  Armenia  Australia  Austria  Bangladesh  \
year                                                                           
2009         NaN     NaN        NaN      NaN        NaN    0.550         NaN   
2010         NaN     NaN      1.904      NaN      2.600    0.295      29.144   
2011         NaN     NaN      1.808      NaN        NaN    0.296         NaN   
2012       2.271     NaN      1.650      NaN        NaN    0.472         NaN   
2013         NaN     NaN      1.619      NaN        NaN    0.000         NaN   

Economy  Belarus  Belgium  Benin  ...  Uganda  Ukraine  United States  \
year                              ...                                   
2009         NaN    1.868    NaN  ...     NaN      NaN            NaN   
2010       0.024    1.358    NaN  ...     NaN      NaN          0.324   
2011       0.062    1.345    NaN  ...     NaN      NaN          0.277   
2012       0.028    0.914    NaN  ...  47.914      NaN          0.298   
2013       0.013    1.407    NaN  ...     NaN      NaN          0.302   

Economy  Uruguay  Vanuatu  Viet Nam  West Bank and Gaza  Yemen, Rep.  Zambia  \
year                                                                           
2009         NaN      NaN       NaN                 NaN          NaN     NaN   
2010       4.388   18.015     6.667               3.764          NaN     NaN   
2011       3.506      NaN       NaN               3.226          NaN     NaN   
2012       3.119      NaN     6.642                 NaN          NaN     NaN   
2013       3.022      NaN       NaN                 NaN          NaN     NaN   

Economy  Zimbabwe  
year               
2009          NaN  
2010          NaN  
2011          NaN  
2012          NaN  
2013          NaN  

[5 rows x 149 columns]

# **10. Graphs**
*texto en cursiva*

In [35]:
import plotly.express as px

## 10.1 Line graph ##


In [50]:
Argentina = d2.query("Economy == 'Argentina'")
Argentina.head()
Austria = d2.query("Economy == 'Austria'")

In [37]:
# fig 1
fig1 = px.line(Argentina, x ="year_str", y = "multidimensional ratio",
               title="my beautiful grafic")
fig1.show()

## 10.1.1 Conslusions one line graph ##

<ul>
  <li>We can see that on this grafic the only evaluated country is Argentina having the lowest multimentional ratio metric in 2017 searching information regarding what could happened nothing seems to be the root cause of this what we do know is that Argentina was not the only LATAM country affected on this specific year </li>
  <li> As a trend Argentina catch up with the rest of the countries having the highest multidimentional ration between 2019 and 2020 againts all forecast
  <li>We can conclude that due to the pandemic the data was even more necessary and for that reason we have a drastic increse on this metric this year, because of the data that was analyzed on this years, after that a negative trend line has been identified.  </li>
</ul>

In [38]:
# fig 1
fig2 = px.line(Argentina, x ="year_str", y = "multidimensional ratio")

#title
fig2.update_layout(title_text = " Beautiful grafic ",
                   title_font_size  = 20,
                   title_x = 0.5)
# axis
fig2.update_xaxes(title = "year", title_font_size = 20)
fig2.update_yaxes(title = "Multidimensional ratio", title_font_size = 20)
fig2.show()

## 10.2 bar graph ##


In [72]:
# fig 1
fig3 = px.bar(Argentina, x ="year_str", y = "multidimensional ratio")

#title
fig3.update_layout(title_text = " Multidimensional ratio in bars Latam countries ",
                   title_font_size  = 20,
                   title_x = 0.5)
# axis
fig3.update_xaxes(title = "year", title_font_size = 20)
fig3.update_yaxes(title = "Multidimensional ratio", title_font_size = 20)
fig3.show()

## 10.2.1 bar graph conclusions ##

<ul>
  <li>Something really interesting is that we can see that the pandemic years were the years in which the general multidimentional ratio was higher than other years against all forecast  </li>
  <li> In 2017 the world experienced the lowest metric meaning that this specific countries of LATAM were affected
  <li>there is an standard metrics we are able to see that this metric is not usual to change drastically owerver the trend for this comming years is not positive since the 2020   </li>
</ul>

## 10.3 more than one line ##

In [59]:
## create a list of countries
c_list = ['Colombia','Chile','Argentina','Bolivia']

d3 = d2.reset_index()
Comparation = d3[d3['Economy'].isin(c_list)]

Comparation = Comparation.sort_values(by=['Economy','year'])
Comparation.tail(10)

Economy  year region Country code  Survey year  monetary    ea    ee  \
59  Colombia  2012    LAC          COL         2012     6.715 7.657 3.549   
60  Colombia  2013    LAC          COL         2013     6.112 7.169 2.963   
61  Colombia  2014    LAC          COL         2014     5.431 6.651 3.028   
62  Colombia  2015    LAC          COL         2015     4.925 6.429 2.862   
63  Colombia  2016    LAC          COL         2016     4.914 6.001 2.690   
64  Colombia  2017    LAC          COL         2017     4.339 5.725 2.542   
65  Colombia  2018    LAC          COL         2018     4.498 5.467 2.618   
66  Colombia  2019    LAC          COL         2019     5.285 5.067 2.837   
67  Colombia  2020    LAC          COL         2020     7.283 5.410 2.902   
68  Colombia  2021    LAC          COL         2021     6.614 4.417 3.130   

    electricity  sanitation  dwater  multidimensional ratio year_str  
59        2.946      13.149   2.546                   8.091     2012  
60        2.434      12.002   2.568                   7.228     2013  
61        2.369      11.818   2.244                   6.414     2014  
62        1.892      10.673   2.197                   5.802     2015  
63        1.642       8.892   2.237                   5.631     2016  
64        1.541       8.717   2.286                   5.028     2017  
65        1.481       8.349   2.493                   5.145     2018  
66        1.321       8.167   2.444                   5.853     2019  
67        1.531       8.765   3.020                   7.818     2020  
68        1.010       7.421   1.727                   6.972     2021

In [42]:
fig4 = px.line(Comparation, x = 'year_str', y = 'multidimensional ratio', color = 'Economy')
fig4.show()

## 10.3.1 More than one line conclusions ##

<ul>
  <li>As part of one of the best countries on this rank que can say that Bolivia as a country has some of the best conditions for the people to leave as an average of the variables studied</li>
  <li> Colombia our country according to the chart is above countries in the first world such as United states Russia, that means or we can conclude that people in our country has more access to Drinking water, electricity o Education.
  <li>Within the four countries analyzed we can see that all of them are from Latam, even that, those countries represent more than the 50 % of the metric for multidimensional ratio meaning in LATAM despite being a third world countries we have higher expectations and a more comfortable way of living   </li>
</ul>

In [73]:
# fig 1
fig3 = px.bar(Argentina, x ="year_str", y = "multidimensional ratio")

#title
fig3.update_layout(title_text = " ultidimensional ratio in bars Latam countries ",
                   title_font_size  = 20,
                   title_x = 0.5)
# axis
fig3.update_xaxes(title = "year", title_font_size = 20)
fig3.update_yaxes(title = "Multidimensional ratio", title_font_size = 20)
fig3.show()

In [44]:
fig5 = px.bar(Comparation, x = 'year', y = 'multidimensional ratio', color = 'Economy')
fig5.show()

## 10.4. pie chart ##

In [60]:
## the idea is to graph the top 10 most populated country

y2009_2 = y2009.sort_values("multidimensional ratio", ascending = False)
y2009_2.head(10)

y2009_2 =y2009_2.iloc[0:20]

In [46]:
# by year
y2020 =d2.query("year == 2020")
y2020.head()

region Country code  Survey year  monetary     ea    ee  \
Economy    year                                                           
Argentina  2020    LAC          ARG         2020     1.075  1.360 0.659   
Armenia    2020    ECA          ARM         2020     0.352  0.057 1.937   
Bolivia    2020    LAC          BOL         2017     3.122 13.976 2.058   
Colombia   2020    LAC          COL         2020     7.283  5.410 2.902   
Costa Rica 2020    LAC          CRI         2020     2.291  3.399 0.510   

                 electricity  sanitation  dwater  multidimensional ratio  \
Economy    year                                                            
Argentina  2020        0.000       0.820   0.124                   1.076   
Armenia    2020        0.000       0.758   1.592                   0.352   
Bolivia    2020        4.372      17.715   6.554                   6.554   
Colombia   2020        1.531       8.765   3.020                   7.818   
Costa Rica 2020        0.105       1.385   0.090                   2.336   

                year_str  
Economy    year           
Argentina  2020     2020  
Armenia    2020     2020  
Bolivia    2020     2020  
Colombia   2020     2020  
Costa Rica 2020     2020

In [71]:
y2020_2 = y2020.sort_values("multidimensional ratio", ascending = False)
y2020_2.head(10)

y2020_2 =y2020_2.iloc[0:10]
y2020.head()

fig5 = px.pie(y2020_2, values = "electricity", names = y2020_2.index.get_level_values(0))
# title
fig5.update_layout(title_text = 'Top 10 Countries with the best Multidimensional ratio - 2020',
                   title_font_size = 20,
                   title_x = 0.5)
fig5.update_traces(textinfo='label+percent', textposition='inside', showlegend=False)

fig5.show()

## 10.4.1 pie chart Conclusion ##


<ul>
  <li>The pie chart show us that a few countries dominate the top positions in terms of the multidimensional ratio, which suggests a significant different between the highest-ranked countries and the rest. These leading countries are likely to have significantly better access to electricity Drinking water and other Variables that contribute to a high multidimensional ratio.</li>
  <li>Something really impressive is that 6 of the countries with the best multidimensional ratio are located in south America meaning that despite whats heard about we as Latinos usually have better conditions to live </li>
</ul>